### PR TITLE
Scroll Snap margin and padding examples

### PR DESCRIPTION
**Pull request by rachelandrew**
These are the examples for the scroll snap padding and margin properties, essentially they are all very similar as they are longhand properties for scroll-margin and scroll-padding.

https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Scroll_Snap

### CODE REVIEW

**Comment by rachelandrew** [comment]
I'm not sure that has helped as I rebased the branch before doing so (because in the past not rebasing has caused problems) and I've now got the other commits in there. Let me know if I should just dump this PR and start over, that seems to be most often the best course of action.

**Comment by rachelandrew** [comment]
With regard to 1. I'm not sure whether we should add the examples or not, I don't know when Firefox will be updated for the new implementation.

4. Those files got in when I rebased the PR, I have no idea how to remove them without actually deleting them from the repo. Git and I do not get along.

**Comment by rachelandrew** [comment]
I've been trying to think of a way to add next or previous bits but it is going to look like these are required to make scroll snap work, which isn't ideal. Do we have a way of adding some instruction - it could just be added via the wiki when embedding them but it would be nicer if it were included with the example itself.

File: live-examples/css-examples/scroll-snap/scroll-padding.html [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<section id="example-choice-list" class="example-choice-list large" data-property="scroll-padding">
```
[the remark this answers is not in the document]
Comment (rachelandrew) [reply live-examples/css-examples/scroll-snap/scroll-padding.html]: I can't reproduce that issue Chrome 71, Windows 10.

I don't understand what you mean by your second comment, "I don't understand how you could have padding top and padding bottom." it's a shorthand so can set all 4 values, I don't know why someone might want to, but they can if they need to.

**Comment by chrisdavidmills** [comment]
Sounds like a good solution. I know exactly how frustrating this is, having been in the same situation before.

**Comment by rachelandrew** [comment]
No problem, I'll address the points made in the new PR and detail them so we don't lose any progress.

**Comment by wbamberg** [comment]
> I've been trying to think of a way to add next or previous bits but it is going to look like these are required to make scroll snap work, which isn't ideal. Do we have a way of adding some instruction - it could just be added via the wiki when embedding them but it would be nicer if it were included with the example itself.

Yes, it is tricky to think of something. Maybe worth experimenting and see how it feels to use. I would much prefer to avoid adding guidance in the Wiki as this introduces a strong dependency between the example and the page, so for example if we wanted to move the examples or present them in a different context they would not make sense.

One precedent for including some instruction in the examples are the animation examples, e.g. https://developer.mozilla.org/en-US/docs/Web/CSS/animation-direction, https://developer.mozilla.org/en-US/docs/Web/CSS/animation-iteration-count, where I think it's sometimes not ideal, but it was the best thing we could come up with :).

Also with the `position` example we have instructions for `sticky`: https://developer.mozilla.org/en-US/docs/Web/CSS/position.

>  I know exactly how frustrating this is, having been in the same situation before.

Yes, I've been there too and I'm sure I will be again.

**Comment by rachelandrew** [comment]
New PR https://github.com/mdn/interactive-examples/pull/1280

**Pull request closed without merging**